hecuba_core/src/ArrayDataStore.cpp

```diff
@@ -95,21 +96,21 @@ void ArrayDataStore::store_numpy_into_cas_by_coords(const uint64_t *storage_id,
     uint32_t half_int = 0;//(uint32_t)-1 >> (sizeof(uint32_t)*CHAR_BIT/2); //TODO be done properly
     int32_t cluster_id, block_id;
 
-    SpaceFillingCurve::PartitionGenerator *partitions_it = this->partitioner.make_partitions_generator(metadata, data,
-                                                                                                       coord);
+    SpaceFillingCurve::PartitionGenerator *partitions_it = new ZorderCurveGeneratorFiltered(metadata, data,
+                                                                                            coord);
 
     std::set<int32_t> clusters = {};
     std::list<Partition> partitions = {};
 
-    while (!partitions_it->isDone()) {clusters.insert(partitions_it->computeNextClusterId()); }
-    partitions_it = this->partitioner.make_partitions_generator(metadata, data, {});
+    while (!partitions_it->isDone()) { clusters.insert(partitions_it->computeNextClusterId()); }
+    partitions_it = new ZorderCurveGeneratorFiltered(metadata, data, coord);
```

Comment:
We agreed on adding the method  this->partitioner.make_filtered_partitions_generator(metadata, data, {});

By calling the constructor we force them to use Zorder (this code will not work with other partitioning)